ACD-502 (#26)
Actindo-AG/Documentation: DataHub/UserInterface/03_ETLExtensions.md

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| String                           | -                              |
| Text field                       | -                              |
| Float                            | -                              |
| Integer                          | -                              |
| Boolean                          | -                              |
| Double float                     | -                              |
| Country                          | -                              |
| Currency                         | -                              |
| Language                         | -                              |
| Tree node                        | -                              |
| Date field                       | -                              |

### Configuration

- *Destination attribute name*   
   Enter a value that is mapped to the destination attribute.

> ### Constant value attribute

Reviewer: H3 oder H2?

![Constant value attribute](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/ConstantValueAttributeValue.png "[Constant value attribute]")

In addition to the *Constant value* extension, several extensions allow to set a constant value as a source attribute. In this case, the mapped data type of the constant value attribute is included in the source attribute name.  

> By clicking the ![Edit](../../Assets/Icons/Edit02.png "[Edit]") (Edit) button within the source attribute the *Enter value* window is displayed to enter the fixed value for the source attribute.

Reviewer: ```suggestion
By clicking the ![Edit](../../Assets/Icons/Edit02.png "[Edit]") (Edit) button within the source attribute, the *Enter value* window is displayed to enter the fixed value for the source attribute.
```

| Text field                       | Country                        |
| Tree node                        | Country                        |

### Configuration

This ETL extension has no further configuration settings.

## Date Converter

![Date Converter](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/DateConverter.png "[Date Converter]")

This extension is used to map a date to a destination attribute while changing its format or modifying the date. The destination date format and additional modifiers to change the date by a specific time value are defined in the configuration.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type                 |
|----------------------------------|--------------------------------------------|
| String                           | Date time, Date field                      |
| Text field                       | Date time, Date field                      |
| Date time                        | Date time, Date field, String, Text field  |                   
| Date field                       | Date time, Date field, String, Text field  |  

### Configuration

> - *Destination Format*   

Reviewer: ```suggestion
- *Destination format*   
```
Groß-/Kleinschreibung sehr gemischt hier, wie üblich, z.B. direkt oben Date Converter, generell viele Header/Extensions... Vereinheintlichen?

  Enter the destination format of the date, for instance **Y-m-d**. The date must be entered in the php format, see https://www.php.net/manual/de/datetime.format.php.

- *Modifier 1*   
  Enter a value for the modifier to change the source date by a certain time value. For instance, enter **+ 7 days** to predate the date in the source attribute always by 7 days in the destination attribute. You can predate a date by using a plus sign, or backdate a date by using a minus sign. You can only enter one modification per modifier. To add further modifications, add another modifier.

- ![Add](../../Assets/Icons/Plus03.png "[Add]") (Add)   
  Click this button to add another modifier. Enter a further value for the modifier to change the source date by a certain time value. The modifier is additionally applied to the source date. For instance, if you enter **+ 1 month** in the first modifier and **+ 1 day** in the second modifier, the destination date is predated by 1 month and 1 day. You can add an unlimited number of modifiers.

> ## EAN to Sku

Reviewer: ```suggestion
## EAN to SKU
```

|----------------------------------|--------------------------------|
| String                           | String                         |

### Configuration

This ETL extension has no further configuration settings.

## Entity Id to Generic Property Extension

![Entity ID to generic property extension](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/EntityIDToGenericPropertyExtension.png "[Entity ID to generic property extension]")

This extension is used to map the generic property of an entity attribute to a destination attribute. This mapping is often used for the data export.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| String                           | String, Text field, Integer    |
| Boolean                          | String, Text field, Integer    |

[comment]: <> (aktueller Bug: Configuration wird nur bei boolean als destination attribut angezeigt. Anwendungsfall aufzeigen! Mehr Infos)

### Configuration

- *Empty drop-down list*   
  Select the appropriate drop-down list ???
  - **Beleg**
  - **Beleg-Position**
  - **Kunde**

> 

Reviewer: Info fehlt. IU Elemente noch zu übersetzen.

- *Field Name*   
  Enter the name of the field assigned to the entity attribute whose value should be mapped to the destination attribute.

> - ![Toggle](../../Assets/Icons/Toggle.png "[Toggle]") *Invert Bool*

Reviewer: ```suggestion
- ![Toggle](../../Assets/Icons/Toggle.png "[Toggle]") *Invert Bool*   
```

This extension is used to map a product bundle attribute to a UCS bundle attribute. This extension is Actindo specific and is only used in the mapping from the PIM to the RetailSuite. It is mainly used to manage the stock of sets.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| UCS bundle product               | Product bundle                 |

### Configuration
This ETL extension has no further configuration settings.

## Html-Template

![HTML template](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/HTMLTemplate.png "[HTML template]")

This extension is used to map an HTML template to a destination attribute. The HTML template to be applied is selected in the configuration. This extension is selected for mappings requiring more logical assignments which cannot be mapped with the other extensions, for instance it is often used for the document export. This extension is only available when the *HTML Templates for ETL* plugin is installed.  

> For detailed information about HTML templates, see [Manage the HTML templates](../../PIM/Operation/03_ManageHTMLTemplates.md).

Reviewer: ```suggestion
For detailed information about HTML templates, see [Manage the HTML templates](../../DataHub/Operation/03_ManageHTMLTemplates.md).
```

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| Text field                       | -                              |
| String                           | -                              |

### Configuration

- *Html-Template*   
  Select the HTML template to be applied to the mapping. The HTML templates can be created in the *HTML TEMPLATES* tab.

## Identity-Mapping

![Identity mapping](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/IdentityMapping.png "[Identity mapping]")

This extension is used to copy the value from the source attribute and map it to the destination attribute. In contrast to the *Basic Mapping* extension, only completely identical attributes in the source and the destination attribute can be mapped.

### Possible data type mappings

> This extension is mostly data type independent. All attributes with a selfmappable data type can be mapped.

Reviewer: ```suggestion
This extension is mostly data type independent. All attributes with a self-mappable data type can be mapped.
```

## Import special price

![Import special price](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/ImportSpecialPrice.png "[Import special price]")

This extension is used to map a promotion price to the PIM price field attribute. The PIM price field is Actindo specific and contains not only the simple price, but also the base price, scale prices, promotion prices, the tax class including the tax rate and supported currencies. The promotion price to be mapped includes several source attributes. By the source attributes, define:  
  - whether the discount is a percent discount or an absolute discount
  - the amount of the discount
  - the currency
  - the start quantity from which the promotion price is applied
  - the start date of the promotion price
  - the end date of the promotion price

Beside the dates, all attributes are required for the mapping.

[comment]: <> (Is that right?)

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|

> | Price field                      | *Percent discount*: Boolean <br/> *Discount percent/absolute special price*: Float, Integer, String <br/> *Currency*: String, Currency <br/> *Start Quantity*: Float, Integer, String <br/> *Promotion Start*: String br/> *Promotion End*: String |

Reviewer: ```suggestion
| Price field                      | *Percent discount*: Boolean <br/> *Discount percent/absolute special price*: Float, Integer, String <br/> *Currency*: String, Currency <br/> *Start Quantity*: Float, Integer, String <br/> *Promotion Start*: String <br/> *Promotion End*: String |
```

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| String                           | Language                       |
| Text field                       | Language                       |

### Configuration

- *languageValue*   
  Select the appropriate display type for the language attribute. The following types are available:
  - **3 digit code**: The three-letter language code according to ISO 639-2 is used. For languages with a bibliographic (B) and a terminological (T) code, both codes are indicated, for instance *ger (B), deu (T)*.
  - **2 digit code**: The two-letter language code according to ISO 639-1 is used, for instance *de*.
  - **Name (english)**: The english language name is used, for instance *german*.
  - **Name (french)**: The french language name is used, for instance *allemand*.
  - **Name (german)**: The german language name is used, for instance *deutsch*.

## Mapping Table

![Mapping table](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/MappingTable.png "[Mapping table]")

> This extension is used to map a set of source attributes to fixed destination attributes. The destination values are assigned in the configuration. This extension is often used in the order import, for instance with different shipping providers names in a web shop and Actindo.

Reviewer: ```suggestion
This extension is used to map a set of source attributes to fixed destination attributes. The destination values are assigned in the configuration. This extension is often used in the order import, for instance with different shipping provider names in a web shop and Actindo.
```

[comment]: <> (SpecialBasePriceToFloatExtension)

![Next promotion base price to float](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/NextPromotionBasePriceToFloat.png "[Next special base price to date time]")

This extension is used to map the promotion price of a PIM price field attribute to a destination attribute. The PIM price field is Actindo specific and contains not only the simple price, but also the base price, scale prices, promotion prices, the tax class including the tax rate and supported currencies. Each of this data is extracted from the price attribute by the corresponding extension. In this case, the promotion price is mapped to the destination attribute. Further settings are defined in the configuration.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type      |
|----------------------------------|---------------------------------|
| Float                            | Price field                     |

### Configuration

- *Country*     
  Select the appropriate country in the drop-down list to apply the correct tax rate to the price. All available countries are displayed in the list.

- *Currency*   

>   Select the appropriate currency in the drop-down list. Only those currencies that are defined as a supported currency in the PIM price field attribute configuration are mapped. in All available currencies are displayed in the list.

Reviewer: ```suggestion
  Select the appropriate currency in the drop-down list. Only those currencies that are defined as a supported currency in the PIM price field attribute configuration are mapped. All available currencies are displayed in the list.
```

## Next special base price to date time

[comment]: <> (SpecialBasePriceToDateTimeExtension)

![Next special base price to date time](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/NextSpecialBasePriceToDateTime.png "[Next special base price to date time]")

This extension is used to map the promotion date of a PIM price field attribute to a destination attribute. The PIM price field is Actindo specific and contains not only the simple price, but also the base price, scale prices, promotion prices, the tax class including the tax rate and supported currencies. Each of this data is extracted from the price attribute by the corresponding extension. In this case, the date of the next promotion is mapped to the destination attribute. Further settings are defined in the configuration.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type      |
|----------------------------------|---------------------------------|
| Date time                        | Price field                     |

### Configuration

- ![Toggle](../../Assets/Icons/Toggle.png "[Toggle]") *Get from date*   

>   Enable the toggle to map the start date of the promotion. Otherwise the end date of the promotion is mapped to the destination attribute. By default, the toggle is disabled.

Reviewer: ```suggestion
  Enable the toggle to map the start date of the promotion. Otherwise, the end date of the promotion is mapped to the destination attribute. By default, the toggle is disabled.
```

### Possible data type mappings

| Destination attribute data type  | Source attribute data type                    |
|----------------------------------|-----------------------------------------------|
| String                           | String, Text field, Integer, Float, Tree node |
| Text field                       | String, Text field, Integer, Float, Tree node |
| Integer                          | Integer                                       |
| Float                            | Integer, Float                                |

### Configuration

- *Relevant Language*   
  Select the language of the attribute value used for the mapping. This setting only applies when a tree node attribute is mapped to a single language attribute. All active languages are displayed in the drop-down list.

## Number-To-String

![Number to String](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/NumberToString.png "[Number to String]")

> This extension is used to map a number to text. Both, absolute and decimal number values are allowed in the source attribute. The value is mapped unchanged, only the data type of the attribute changes.

Reviewer: ```suggestion
This extension is used to map a number to text. Both absolute and decimal number values are allowed in the source attribute. The value is mapped unchanged, only the data type of the attribute changes.
```

### Possible data type mappings

| Destination attribute data type  | Source attribute data type      |
|----------------------------------|---------------------------------|
| String                           | Integer, Float                  |
| Text field                       | Integer, Float                  |

### Configuration

This ETL extension has no further configuration settings.

## Number-To-UnitValue

![Number to unit value](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/NumberToUnitValue.png "[Number to unit value]")

> This extension is used to map a number to a unit attribute. Both, absolute and decimal number values are allowed in the source attribute. A unit attribute always includes a quantity and a unit value. This mapping is used to map a source attribute, which is maintained in a certain unit and therefore does not include the unit in the source attribute, to a unit attribute. Define the unit settings in the configuration.

Reviewer: ```suggestion
This extension is used to map a number to a unit attribute. Both absolute and decimal number values are allowed in the source attribute. A unit attribute always includes a quantity and a unit value. This mapping is used to map a source attribute, which is maintained in a certain unit and therefore does not include the unit in the source attribute, to a unit attribute. Define the unit settings in the configuration.
```

| Float                            | Base price                     |

### Configuration

This ETL extension has no further configuration settings.

## PIM BasePrice to Boolean (VPE Active)

![PIM base price to boolean](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/PIMBasePriceToBoolean.png "[PIM base price to unit]")

This extension is used to map a PIM base price attribute to a boolean attribute, for instance to define whether or not a base price is available. If the PIM base price attribute is empty, the false value is mapped to the boolean attribute. Otherwise, the true value is mapped to the boolean attribute.

[comment]: <> (Is that right? Anwendungsfall?)

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| Boolean                          | Base price                     |

### Configuration

This ETL extension has no further configuration settings.

## PIM BasePrice to Unit

![PIM base price to unit](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/PIMBasePriceToUnit.png "[PIM base price to unit]")

> This extension is used to map a PIM base price attribute to a tree node attribute. The PIM base price is Actindo specific and contains the dimension, the unit and the relevant amount for the base price. Each of this data is extracted from the PIM base price attribute by the corresponding extension. In this case, the unit is mapped to the destination attribute.

Reviewer: To Tree node or to Unit?

[comment]: <> (Is that right?)

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| Tree node                        | Base price                     |

### Configuration

This ETL extension has no further configuration settings.

## PIM BasePrice to VPE Reference Unit

![PIM base price to VPE reference unit](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/PIMBasePriceToVPEReferenceUnit.png "[PIM base price to VPE reference unit]")

> This extension is used to map a PIM base price attribute to a number. Both, absolute and decimal number values are allowed in the destination attribute. The PIM base price is Actindo specific and contains the dimension, the unit and the relevant amount for the base price. Each of this data is extracted from the PIM base price attribute by the corresponding extension. In this case, the relevant amount is taken and the corresponding reference amount is mapped to the destination attribute. By default, the reference amount equals to 0,1 for all amounts smaller than 0,25 and to the specified amount for all amounts greater than 0,25.

Reviewer: ```suggestion
This extension is used to map a PIM base price attribute to a number. Both absolute and decimal number values are allowed in the destination attribute. The PIM base price is Actindo specific and contains the dimension, the unit and the relevant amount for the base price. Each of this data is extracted from the PIM base price attribute by the corresponding extension. In this case, the relevant amount is taken and the corresponding reference amount is mapped to the destination attribute. By default, the reference amount equals to 0,1 for all amounts smaller than 0,25 and to the specified amount for all amounts greater than 0,25.
```

    - **Consider only promotions**: Only the promotion prices in the PIM price field attribute are mapped to the destination attribute.
    - **Ignore promotions**: All prices beside the promotion prices in the PIM price field attribute are mapped to the destination attribute.

[comment]: <> (Is that correct? Why country? Is a tax class or rate mapped? Where are the promotions prices mapped to: scale prices???)

## Preg Replace Extension

![Preg replace extension](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/PregReplaceExtension.png "[Preg replace extension]")

This extension is used to replace the value of a source attribute by a certain value and map it to a destination attribute. The replacement settings are defined in the configuration.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type         |
|----------------------------------|------------------------------------|
| String                           | String, Text field, Integer, Float |
| Text field                       | String, Text field, Integer, Float |

### Configuration

- *Pattern*   

>   Enter a regular expression for the source attribute value that should be replaced. For detailed information about regular expressions, see https://regex101.com/.

Reviewer: In Taxes haben wir diesen Satz verwendet:  The site https://regex101.com/ can be useful to test regular expressions.

- *Replacement*    
  Enter the value that will replace the value selected in the *Pattern* field.

## PriceBuilder

[comment]: <> (StringsToPriceExtension)

![Price builder](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/PriceBuilder.png "[Price builder]")

> This extension is used to map an several attributes containing price data to a PIM price field attribute. The PIM price field is Actindo specific and contains not only the simple price, but also the base price, scale prices, promotion prices, the tax class including the tax rate and supported currencies. The price to be mapped is composed of different source attributes. By the source attributes, define:  

Reviewer: ```suggestion
This extension is used to map several attributes containing price data to a PIM price field attribute. The PIM price field is Actindo specific and contains not only the simple price, but also the base price, scale prices, promotion prices, the tax class including the tax rate and supported currencies. The price to be mapped is composed of different source attributes. By the source attributes, define:  
```

### Configuration

This ETL extension has no further configuration settings.

## Price to price with discount

![Price to price with discount](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/PriceToPriceWithDiscount.png "[Price to price with discount]")

This extension is used to map a PIM price field attribute including a defined discount to another PIM price field attribute. In the discount attribute, a number for the percentage discount is defined. The source price field attribute less the selected discount is mapped to the destination price attribute.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type                                        |
|----------------------------------|-------------------------------------------------------------------|
| Price field                      | *Price*: Price field </br> *Discount in percent*: Integer, Float  |

### Configuration

This ETL extension has no further configuration settings.

## Price to Tax

![Price to tax](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/PriceToTax.png "[Price to tax]")

> This extension is used to map a PIM price field attribute to a decimal number. The PIM price field is Actindo specific and contains not only the simple price, but also the base price, scale prices, promotion prices, the tax class including the tax rate and supported currencies. Each of this data is extracted from the price attribute by the corresponding extension. In this case, the tax rate is calculated by the tax class set in the attribute which is based on the destination country selected in the configuration and mapped to the destination attribute. The country and the display type of the tax rate are defined in the configuration.

Reviewer: ```suggestion
This extension is used to map a PIM price field attribute to a decimal number. The PIM price field is Actindo specific and contains not only the simple price, but also the base price, scale prices, promotion prices, the tax class including the tax rate and supported currencies. Each of this data is extracted from the price attribute by the corresponding extension. In this case, the tax rate is calculated by the tax class set in the attribute, which is based on the destination country selected in the configuration, and mapped to the destination attribute. The country and the display type of the tax rate are defined in the configuration.
```

[comment]: <> (Is that right?)

## Price to Tax Class

![Price to tax class](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/PriceToTaxClass.png "[Price to tax class]")

This extension is used to map a PIM price field attribute to a destination attribute. The PIM price field is Actindo specific and contains not only the simple price, but also the base price, scale prices, promotion prices, the tax class including the tax rate and supported currencies. Each of this data is extracted from the price attribute by the corresponding extension. In this case, the tax class is mapped to the destination attribute.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| Tax class                        | Price field                    |
| Tree node                        | Price field                    |

### Configuration

The configuration settings are only displayed when a destination attribute with the tree node data type is selected. A single drop-down list is displayed for each tax class defined in Actindo.

> **Map PIM-Taxclasses to corresponding Tree Nodes**

Reviewer: ```suggestion
**Map PIM tax classes to corresponding Tree Nodes**
```

- *Standardsatz*   
  Select the tax rate to be mapped to the standard rate in the destination attribute. All available tax rates from the destination attribute are displayed in the drop-down list.

- *Ermäßigt*   
  Select the tax rate to be mapped to the reduced rate in the destination attribute. All available tax rates from the destination attribute are displayed in the drop-down list.

[comment]: <> (Stimmt das?)

> 

Reviewer: UI Elemente übersetzen? In Taxes: Standard rate und Reduced rate

### Possible data type mappings

| Destination attribute data type  | Source attribute data type                                              |
|----------------------------------|-------------------------------------------------------------------------|
| Float                            | *Price Field*: Price field </br> *Discount in percent*: Integer, Float  |
| String                           | *Price Field*: Price field </br> *Discount in percent*: Integer, Float  |

### Configuration

- *Country*     
  Select the appropriate country in the drop-down list to apply the correct tax rate to the price. All available countries are displayed in the list.

- *Currency*   

>   Select the appropriate currency in the drop-down list. Only those currencies that are defined as a supported currency in the PIM price field attribute configuration are mapped. in All available currencies are displayed in the list.

Reviewer: ```suggestion
  Select the appropriate currency in the drop-down list. Only those currencies that are defined as a supported currency in the PIM price field attribute configuration are mapped. All available currencies are displayed in the list.
```

This extension is used to map a PIM price field attribute with a fallback attribute, if the first attribute is not available, to a decimal number. This extension is similar to the *Pricing to float* extension with an additional fallback price. The *Special Price* attribute is always considered first when mapping. The *Fallback Price* attribute is only used as a fallback price that will be mapped when no other price is available. The fallback price attribute is mandatory. This mapping is used when at least two PIM price field attributes with different prices are maintained. Further price settings are defined in the configuration.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type                      |
|----------------------------------|-------------------------------------------------|
| Float                            | *Fallback Price*, *Special Price*: Price field  |

### Configuration

- *Country*     
  Select the appropriate country in the drop-down list to apply the correct tax rate to the price. All available countries are displayed in the list.

- *Currency*   

>   Select the appropriate currency in the drop-down list. Only those currencies that are defined as a supported currency in the PIM price field attribute configuration are mapped. in All available currencies are displayed in the list.

Reviewer: ```suggestion
  Select the appropriate currency in the drop-down list. Only those currencies that are defined as a supported currency in the PIM price field attribute configuration are mapped. All available currencies are displayed in the list.
```

  [comment]: <> (Is that right?)

- ![Toggle](../../Assets/Icons/Toggle.png "[Toggle]") *Destination gross*   
  Enable the toggle to map the gross price of the source attribute. Disable the toggle to map the net price.

[comment]: <> (gibt es die extensions RMA-Extension und RMAExtension PosId Extensions noch?
## RMA-Extension ???

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| String                           | *EAN*, *Menge*: String         |

### Configuration

## RMAExtension PosId Extensions ???

### Possible data type mappings

| Destination attribute data type  | Source attribute data type        |
|----------------------------------|-----------------------------------|
| String                           |*Projekt Nummer*, *EAN*: String    |

### Configuration )

## Simple-Pricing-To-Float

![Simple pricing to float](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/SimplePricingToFloat.png "[Simple pricing to float]")

> This extension is used to map a simple price attribute to a decimal number. The simple price attribute is Actindo specific and therefore needs a special extension for mapping. As the simple price attribute is not often used also the extension is rarely needed.  

Reviewer: ```suggestion
This extension is used to map a simple price attribute to a decimal number. The simple price attribute is Actindo specific and therefore needs a special extension for mapping. As the simple price attribute is not often used, also the extension is rarely needed.  
```

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| Float                            | Simple pricing                 |

### Configuration

> - *Empty Value*

Reviewer: ```suggestion
- *Empty Value*   
```

### Possible data type mappings

| Destination attribute data type  | Source attribute data type                      |
|----------------------------------|-------------------------------------------------|
| Price field                      | *Fallback price*, *Special price*: Price field  |

### Configuration

- *Country*     
  Select the appropriate country in the drop-down list to apply the correct tax rate to the price. All available countries are displayed in the list.

- *Mapping mode*   
  Select the appropriate mapping mode for the source price attributes. The following modes are available:   
    - **Consider all prices**: All prices in the PIM price field attribute are mapped to the destination attribute.
    - **Consider only promotions**: Only the promotion prices in the PIM price field attribute are mapped to the destination attribute.
    - **Ignore promotions**: All prices beside the promotion prices in the PIM price field attribute are mapped to the destination attribute.

## Stock-To-Integer

![Stock to integer](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/StockToInteger.png "[Stock to integer]")

> This extension is used map a stock attribute to an absolute number. The stock attribute is Actindo specific and therefore needs a special extension for mapping. This extension is mainly used to map the stock from Actindo to another system.  

Reviewer: ```suggestion
This extension is used to map a stock attribute to an absolute number. The stock attribute is Actindo specific and therefore needs a special extension for mapping. This extension is mainly used to map the stock from Actindo to another system.  
```

[comment]: <> (ETLTreeExtension)

![String tree to string tree](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/StringTreeToStringTree.png "[String tree to string tree]")

This extension is used to either map a source attribute to a tree node attribute or a tree node attribute to a text. This extension is often used for csv imports where only data input as string data type is allowed but the data should result in a tree node data type.  

### Possible data type mappings

| Destination attribute data type  | Source attribute data type     |
|----------------------------------|--------------------------------|
| String                           | Tree node                      |
| Float                            | Tree node                      |
| Integer                          | Tree node                      |
| Text field                       | Tree node                      |
| Tree node                        | String, Text field             |

### Configuration

- *Separator*   
  Enter a separator sign by which a lower level in the string attribute is indicated. For instance, to indicate several gradations for a color (Red|Coral).

- *Destination Field*   

>   Select the field of the tree node attribute whose value will be used for the mapping. The drop-down down list is only displayed if the destination attribute is a tree node attribute.

Reviewer: ```suggestion
  Select the field of the tree node attribute whose value will be used for the mapping. The drop-down list is only displayed if the destination attribute is a tree node attribute.
```

  [comment]: <> (Anwendungsbeispiel für relevant locale? Verstehe den Einsatz nicht so richtig...)

## String and Value to Unit

![String and value to unit](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/StringValueToUnit.png "[String and value to unit]")

This extension is used to map two attributes including a unit and a unit value to a unit attribute. A unit attribute always includes a unit and a quantity. The unit to be mapped is defined in the *Unit* attribute, the quantity in the *Value* attribute.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type                            |
|----------------------------------|-------------------------------------------------------|
| Unit                             | *Unit*: String </br> *Value*: String, Float, Integer  |

### Configuration

This ETL extension has no further configuration settings.

## String Date Converter

![String date converter](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/StringDateConverter.png "[String date converter]")

> This extension is used to map a date within a string attribute to a text attribute. Both, a string and a text field data type are allowed in the source attribute. The source as well as the destination date format are defined in the configuration. Further, additional modifiers can be defined to change the date by a defined time value.

Reviewer: ```suggestion
This extension is used to map a date within a string attribute to a text attribute. Both a string and a text field data type are allowed in the source attribute. The source as well as the destination date format are defined in the configuration. Further, additional modifiers can be defined to change the date by a defined time value.
```

  Enter the destination format of the date, for instance **Y-m-d**. The date must be entered in the php format, see https://www.php.net/manual/de/datetime.format.php.

- *Modifier 1*   
  Enter a value for the modifier to change the source date by a certain time value. For instance, enter **+ 7 days** to predate the date in the source attribute always by 7 days in the destination attribute. You can predate a date by using a plus sign, or backdate a date by using a minus sign. You can only enter one modification per modifier. To add further modifications, add another modifier.

- ![Add](../../Assets/Icons/Plus03.png "[Add]") (Add)   
  Click this button to add another modifier. Enter a further value for the modifier to change the source date by a certain time value. The modifier is additionally applied to the source date. For instance, if you enter **+ 1 month** in the first modifier and **+ 1 day** in the second modifier, the destination date is predated by 1 month and 1 day. You can add an unlimited number of modifiers

## String to Image

![String to image](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/StringToImage.png "[String to image]")

> This extension is used to map multiple string attributes to an image attribute. At least one string attribute containing the image url must be selected for the mapping and up to eight source attributes can be mapped. To each image you can define one or several tags. Define the tags separator in the configuration.

Reviewer: ```suggestion
This extension is used to map multiple string attributes to an image attribute. At least one string attribute containing the image URL must be selected for the mapping and up to eight source attributes can be mapped. To each image you can define one or several tags. Define the tags separator in the configuration.
```

|----------------------------------|-----------------------------|
| String                           | String                      |

### Configuration

- *Glue*   
  Enter a connector or separator which is added between the different source attribute values.

## String-To-Absolute-Number

![String to absolute number](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/StringToAbsoluteNumber.png "[String to absolute number]")

This extension is used to map a string attribute to a number. The value is mapped unchanged, only the data type of the attribute changes. This mapping is used for absolute numbers. To map decimal numbers, use the *String-To-Number* extension.

[comment]: <> (What happens if a string attribute contains a floating number? Is it rounded? And why is the float data type allowed as Destination?)

### Possible data type mappings

| Destination attribute data type  | Source attribute data type  |
|----------------------------------|-----------------------------|
| Integer                          | String                      |
| Float                            | String                      |

### Configuration

> - *separator*

Reviewer: ```suggestion
- *separator*   
```

  Select the appropriate separator for decimal numbers. The following signs are available:  
  - **,**: By default, the comma is used as a decimal separator in German.
  - **.**: By default, the point is used as a decimal separator in English.

  [comment]: <> (separator for absolute numbers? where is an absolute number separated?)

> - *emptyString*

Reviewer: ```suggestion
- *emptyString*   
```

  Select what will be mapped to the destination attribute if the source attribute value is empty. The following options are available:

>   - **leer**: The destination attribute value is empty.

Reviewer: ```suggestion
  - **empty**: The destination attribute value is empty.
```
? Oder null?

This extension is used to map a string attribute to a boolean attribute. If the string attribute value equals 0, the false value is mapped to the boolean attribute. If the string attribute value equals 1, the true value is mapped to the boolean attribute.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type  |
|----------------------------------|-----------------------------|
| Boolean                          | String                      |

### Configuration

This ETL extension has no further configuration settings.

## String-To-Country

![String to country](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/StringToCountry.png "[String to country]")

This extension is used to map a text to a country attribute. This mapping is mainly used for the import to Actindo as the country attribute is Actindo specific. Define how the country is indicated in the configuration.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type  |
|----------------------------------|-----------------------------|
| Country                          | String, Text field          |

### Configuration

> - *fieldToSearch*

Reviewer: ```suggestion
- *fieldToSearch*   
```

  Select the appropriate information about how the country is entered in the source attribute, so that the system can map the country correctly to the destination attribute. The following options are available:
  - **Numerischer Ländercode**: Three-digit country code according to ISO 3166-1 numeric, for instance *840*.
  - **Dreibuchstabiger Ländercode**: Three-letter country code according to ISO 3166-1 alpha-3, for instance *USA*.
  - **Zweibuchstabiger Ländercode**: Two-letter country code according to ISO 3166-1 alpha-2, for instance *US*.
  - **Kfz Ländercode**: International vehicle registration code, for instance *USA*.
  - **Name**: Country name (in the national language), for instance *United States*.
  - **FusionMaps Ländercode**: ???

> 

Reviewer: UI Elemente zu übersetzen?

  [comment]: <> (Was ist FusionMaps?)

## String-To-Number

![String to number](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/StringToNumber.png "[String to number]")

> This extension is used to map a string attribute to a number. The value is mapped unchanged, only the data type of the attribute changes. In contrast to the *String-To-Absolute-Number* extension, both, decimal and absolute numbers can be mapped to the destination attribute.

Reviewer: ```suggestion
This extension is used to map a string attribute to a number. The value is mapped unchanged, only the data type of the attribute changes. In contrast to the *String-To-Absolute-Number* extension, both decimal and absolute numbers can be mapped to the destination attribute.
```

### Possible data type mappings

| Destination attribute data type  | Source attribute data type  |
|----------------------------------|-----------------------------|
| Integer                          | String                      |
| Float                            | String                      |

### Configuration

> - *separator*

Reviewer: ```suggestion
- *separator*   
```

  Select the appropriate separator for decimal numbers. The following signs are available:  
  - **,**: By default, the comma is used as a decimal separator in German.
  - **.**: By default, the point is used as a decimal separator in English.

> - *emptyString*

Reviewer: ```suggestion
- *emptyString*   
```

|----------------------------------|-----------------------------|
| String                           | Tax zone                    |

### Configuration

This ETL extension has no further configuration settings.

## Text-To-UnitValue

![Text to unit value](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/TextToUnitValue.png "[Text to unit value]")

This extension is used to map a text to a unit attribute. A unit attribute always includes a unit and a quantity. Only the quantity value is mapped by the source attribute. The default unit to the destination dimension is automatically used as unit for the destination attribute.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type   |
|----------------------------------|------------------------------|
| Unit                             | String, Text field           |

### Configuration

This ETL extension has no further configuration settings.

## Tree and Number to base price

![Tree and number to base price](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/TreeNumberToBasePrice.png "[Tree and number to base price]")

> This extension is used to map a tree node attribute and a number to a PIM base price attribute. The base price is Actindo specific and contains the dimension, the unit and the relevant amount for the base price. The amount is for the base price is mapped by the *Value* attribute, the unit by the *Unit* attribute. The dimension and the unit are defined in the configuration. The configuration settings are only displayed when a unit attribute is selected.

Reviewer: ```suggestion
This extension is used to map a tree node attribute and a number to a PIM base price attribute. The base price is Actindo specific and contains the dimension, the unit and the relevant amount for the base price. The amount for the base price is mapped by the *Value* attribute, the unit by the *Unit* attribute. The dimension and the unit are defined in the configuration. The configuration settings are only displayed when a unit attribute is selected.
```

[comment]: <> (Is that right?)

### Possible data type mappings

| Destination attribute data type  | Source attribute data type                          |
|----------------------------------|-----------------------------------------------------|
| Price field                      | *Price*: Simple price </br> *Tax class*:  Tree node |

### Configuration

The configuration depends on the selected tax class attribute. A single drop-down list is displayed for each tax class defined in the source system.

> **Assign tax classes for attribute**

Reviewer: ```suggestion
**Assign tax classes for attribute "Name"**
```
?

- *Reduced rate 2*   
  Select the tax rate to be mapped to the reduced rate 2 in the destination attribute. All available tax rates from the destination attribute are displayed in the drop-down list.

- *Reduced rate*   
  Select the tax rate to be mapped to the reduced rate in the destination attribute. All available tax rates from the destination attribute are displayed in the drop-down list.

- *Standard rate*   
  Select the tax rate to be mapped to the standard rate in the destination attribute. All available tax rates from the destination attribute are displayed in the drop-down list.

## Tree(Key)-To-String

![TreeKey to String](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/TreeKeyToString.png "[TreeKey to String]")

> This extension is used to map a tree node attribute to text. Both, string and a text field attributed are allowed in the destination attribute. The key value of the source tree node attribute is mapped to the destination attribute.

Reviewer: ```suggestion
This extension is used to map a tree node attribute to text. Both string and a text field attributed are allowed in the destination attribute. The key value of the source tree node attribute is mapped to the destination attribute.
```

|----------------------------------|-------------------------------------|
| Tree node                        | Tree node                           |

### Configuration

- ![Toggle](../../Assets/Icons/Toggle.png "[Toggle]") *Create node if not existing*   
  Enable the toggle to automatically create a tree node in the destination attribute if no tree node exists. Otherwise, you have to manually create a tree node.

  [comment]: <> (is that right? or does the mapping fail then?)

- *Match destination node by*   
  Select the field of the tree node attribute whose value will be used for the mapping.
  - **Key**: The key value of the tree node attribute is used for the mapping.
  - **Title**: The title value of the tree node attribute is used for the mapping.

- *Relevant Language*   
  Select the language of the attribute value used for the mapping. All active languages are displayed in the drop-down list. The drop-down list is only displayed if the **Title** option is selected in the *Match destination node by* drop-down list.

## Tree value mapping

![Tree value mapping](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/TreeValueMapping.png "[Tree value mapping]")

> This extension is used to map a value connected to a string or tree node attribute to a tree node attribute. The connected value of the source attribute must be accessed to map it to the tree node attribute. For instance, this extension is often used to map the Actindo payment method from an order. All payment methods and their IDs are defined in the *Invoicing* module. These values must be accessed to map both, the payment methods and their IDs to the destination tree node attribute.

Reviewer: ```suggestion
This extension is used to map a value connected to a string or a tree node attribute to a tree node attribute. The connected value of the source attribute must be accessed to map it to the tree node attribute. For instance, this extension is often used to map the Actindo payment method from an order. All payment methods and their IDs are defined in the *Invoicing* module. These values must be accessed to map both, the payment methods and their IDs to the destination tree node attribute.
```

### Possible data type mappings

| Destination attribute data type  | Source attribute data type   |
|----------------------------------|------------------------------|
| Tree node                        | Tree node, String            |

### Configuration

This ETL extension has no further configuration settings.

## UnitValue-To-Number

![Unit value to number](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/UnitValueToNumber.png "[Unit value to number]")

> This extension is used to map an unit attribute to a number. Both, decimal and absolute numbers are allowed in the destination attribute. A unit attribute always includes a unit and a quantity. Only the quantity value is mapped to the destination attribute. To map the quantity value in the correct unit, the quantity is converted into the unit of the destination attribute. Define the unit in the configuration.

Reviewer: ```suggestion
This extension is used to map an unit attribute to a number. Both decimal and absolute numbers are allowed in the destination attribute. A unit attribute always includes a unit and a quantity. Only the quantity value is mapped to the destination attribute. To map the quantity value in the correct unit, the quantity is converted into the unit of the destination attribute. Define the unit in the configuration.
```

> This extension is used to map an unit attribute to a number. Both, decimal and absolute numbers are allowed in the destination attribute. A unit attribute always includes a unit and a quantity. Only the quantity value is mapped to the destination attribute. To map the quantity value in the correct unit, the quantity is converted into the unit of the destination attribute. Define the unit in the configuration.

Reviewer: ```suggestion
This extension is used to map a unit attribute to a number. Both decimal and absolute numbers are allowed in the destination attribute. A unit attribute always includes a unit and a quantity. Only the quantity value is mapped to the destination attribute. To map the quantity value in the correct unit, the quantity is converted into the unit of the destination attribute. Define the unit in the configuration.
```

  - **Suffix Name**: The unit is displayed by its name, for instance *centimeter*.
  - **Suffix Empty**: The unit is hidden.

- *Separator*   
  Enter the separator sign placed between quantity and unit. If the field is left empty, the unit and quantity are written directly one after the other.

- *Decimal Separator*   
  Enter the decimal separator for the quantity value. If the number of decimal places is greater than 0, you should enter a separator sign to avoid that the value is displayed incorrectly. By default, the point is used as a decimal separator in English, the comma in German.

- *Decimal Places*   
  Enter the number of decimal places for the quantity value. This field is mandatory. Depending on the decimal places, the quantity is rounded.

  [comment]: <> (Verwirrend: warum wird bei der unitvalue-to-number erweiterung nur der mengenwert, bei der unitvalue-to-string erweiterung aber menge inklusive einheit übertragen?)

## UnitValue-To-UnitValue

![Unit value to unit value](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/UnitValueToUnitValue.png "[Unit value to unit value]")

> This extension is used to map an unit attribute to a another unit attribute. A unit attribute always includes a unit and a quantity. Define in the configuration which of the units should be mapped in the destination attribute. In contrast to the *Basic Mapping* extension, the *UnitValue-To-UnitValue* extension allows to convert the value when using different units in the source and the destination attribute.

Reviewer: ```suggestion
This extension is used to map a unit attribute to another unit attribute. A unit attribute always includes a unit and a quantity. Define in the configuration which of the units should be mapped in the destination attribute. In contrast to the *Basic Mapping* extension, the *UnitValue-To-UnitValue* extension allows to convert the value when using different units in the source and the destination attribute.
```

## Variant-To-String

![Variant to string](../../Assets/Screenshots/DataHub/Settings/ETL/Extensions/VariantToString.png "[Variant to string]")

This extension is used to map a variant attribute to text, for instance to provide product information. The defining attribute value of the variant is mapped to the destination attribute. Define in the configuration if certain attributes should be excluded and if the attribute names should be included for the mapping.

### Possible data type mappings

| Destination attribute data type  | Source attribute data type          |
|----------------------------------|-------------------------------------|
| String                           | Variant                             |
| Text field                       | Variant                             |

### Configuration

- ![Toggle](../../Assets/Icons/Toggle.png "[Toggle]") *Include attribute name*     

>   Enable the toggle to map both, the attribute names and their values. Disable the toggle to map the attribute values only.

Reviewer: ```suggestion
  Enable the toggle to map both the attribute names and their values. Disable the toggle to map the attribute values only.
```

- ![Toggle](../../Assets/Icons/Toggle.png "[Toggle]") *Automatically add not mapped defining attributes to destination set when creating variant set*   
  Enable the toggle to automatically add unmapped defining attributes to the created destination variant set. It is recommended to enable the toggle. Otherwise, the defining attributes must be added manually to the attribute set. By default, the toggle is enabled. This toggle is only displayed if the *Automatically map variant sets* toggle is enabled.

- ![Toggle](../../Assets/Icons/Toggle.png "[Toggle]") *Sync variant set in case they do not match to each other*   
  Enable the toggle to automatically create a variant set if the assigned variant sets of the source variant attribute and the destination variant attribute do not match. By default, the toggle is disabled. This toggle is only displayed if the *Automatically map variant sets* toggle is disabled.

>   - ![Toggle](../../Assets/Icons/Toggle.png "[Toggle]") *Do not apply master product status to children*      

Reviewer: Eingerückt
